src/scripts/site.js

```diff
@@ -1,216 +1,333 @@
+        window.Events = events;
+        events.on(ConnectionManager, "apiclientcreated", onApiClientCreated);
+
+        connectionManager.currentApiClient = function () {
             if (!localApiClient) {
                 var server = connectionManager.getLastUsedServer();
-                server && (localApiClient = connectionManager.getApiClient(server.Id))
+
+                if (server) {
+                    localApiClient = connectionManager.getApiClient(server.Id);
+                }
             }
-            return localApiClient
-        }, connectionManager.onLocalUserSignedIn = function(user) {
-            return localApiClient = connectionManager.getApiClient(user.ServerId), window.ApiClient = localApiClient, userSettings.setUserInfo(user.Id, localApiClient)
-        }, events.on(connectionManager, "localusersignedout", function() {
-            userSettings.setUserInfo(null, null)
-        })
+
+            return localApiClient;
+        };
+
+        connectionManager.onLocalUserSignedIn = function (user) {
+            localApiClient = connectionManager.getApiClient(user.ServerId);
+            window.ApiClient = localApiClient;
+            return userSettings.setUserInfo(user.Id, localApiClient);
+        };
+
+        events.on(connectionManager, "localusersignedout", function () {
+            userSettings.setUserInfo(null, null);
+        });
     }
 
     function createConnectionManager() {
-        return new Promise(function(resolve, reject) {
-            require(["connectionManagerFactory", "apphost", "credentialprovider", "events", "userSettings"], function(ConnectionManager, apphost, credentialProvider, events, userSettings) {
-                var credentialProviderInstance = new credentialProvider;
-                var promises = [apphost.getSyncProfile(), apphost.init()];
-                Promise.all(promises).then(function(responses) {
-                    var deviceProfile = responses[0],
-                        capabilities = Dashboard.capabilities(apphost);
-                    capabilities.DeviceProfile = deviceProfile;
-                    var connectionManager = new ConnectionManager(credentialProviderInstance, apphost.appName(), apphost.appVersion(), apphost.deviceName(), apphost.deviceId(), capabilities, window.devicePixelRatio);
-                    if (defineConnectionManager(connectionManager), bindConnectionManagerEvents(connectionManager, events, userSettings), !AppInfo.isNativeApp) return console.log("loading ApiClient singleton"), getRequirePromise(["apiclient"]).then(function(apiClientFactory) {
+        return require(["connectionManagerFactory", "apphost", "credentialprovider", "events", "userSettings"], function (ConnectionManager, apphost, credentialProvider, events, userSettings) {
+            var credentialProviderInstance = new credentialProvider(),
+                promises                   = [apphost.getSyncProfile(), apphost.init()];
```

Comment:
I'm not a fan of this. I'd really prefer having variable declarations on separate lines.